eeps/eep-0072.md

```diff
@@ -0,0 +1,145 @@
+    Author: Jesse Gumm <ja(at)gumm(dot)io>
+    Status: Draft
+    Type: Standards Track
+    Created: 08-Oct-2024
+    Post-History:
+****
+EEP 72: Reserved words and Variables as record names, and enhancement to definition syntax
+----
+
+Abstract
+========
+
+This EEP loosens some of the restrictions around record names such no longer is
+it necessary to quote record names when those records are named with reserved
+words (`#if` vs `#'if'`) or words with capitalized first characters (`#Hello`
+vs `#'Hello'`).
+
+This EEP also proposes to add a new record-like syntax to the record
+definitions (also adopting the above syntactical changes), such that the
+following record definitions would all be valid and identical:
+
+```erlang
+-record('div', {a :: integer(), b :: integer()}).
+-record #'div'{a :: integer(), b :: integer()}.
+-record #div{a :: integer(), b :: integer()}.
+```
+
```

Comment:
````suggestion
The latter two are proposed new syntax.  The following would also be valid
and identical since parentheses are optional in attributes:

```erlang
-record 'div', {a :: integer(), b :: integer()}.
-record(#'div'{a :: integer(), b :: integer()}).
-record(#div{a :: integer(), b :: integer()}).
```
````